src/openqaoa-core/algorithms/rqaoa/rqaoa_utils.py

```diff
@@ -641,47 +619,33 @@ def solution_for_vanishing_instances(hamiltonian: Hamiltonian, spin_map: dict):
     Returns
     -------
     cl_energy: `float`
-        The energy of the first solution wrt the cost Hamiltonian.
+        The energy of the solution wrt the cost Hamiltonian.
     cl_ground_states: `list`
-        List of strings of binary values representing the classical solution of the
-        problem respecting the spin map.
+        List of a string of binary values representing the classical solution of the
+        problem in correspondance with the spin map.
     """
-    cl_ground_states = [""]
+    cl_ground_state = ""
 
     for spin in spin_map.keys():
-        new_cl_ground_states = []
-
         if spin_map[spin][1] == None:
-            # add 0 or 1 arbitrarily
-
-            for ground_state in cl_ground_states:
-                first_new_ground_state = ground_state + "0"
-                second_new_ground_state = ground_state + "1"
-
-                new_cl_ground_states.append(first_new_ground_state)
-                new_cl_ground_states.append(second_new_ground_state)
-
-            cl_ground_states = new_cl_ground_states
-
+            cl_ground_state += "1"
         else:
             # fix according to correlation factor
             factor = spin_map[spin][0]
             parent = spin_map[spin][1]
 
-            for ground_state in cl_ground_states:
-                if factor == 1.0:
-                    # correlated
-                    new_value_spin = ground_state[parent]
-                else:
-                    # anticorrelated
-                    new_value_spin = str(int(not bool(int(ground_state[parent]))))
-
-                new_ground_state = ground_state + new_value_spin
-                new_cl_ground_states.append(new_ground_state)
-
-            cl_ground_states = new_cl_ground_states
+            if factor == 1.0:
+                # correlated
+                cl_ground_state += cl_ground_state[parent]
+            else:
+                # anticorrelated
+                cl_ground_state += str(int(not bool(int(cl_ground_state[parent]))))
 
     # computing the energy of the first one only, assuming degeneracy
-    cl_energy = bitstring_energy(hamiltonian, cl_ground_states[0])
+    cl_energy = bitstring_energy(hamiltonian, cl_ground_state)
+
+    # to match the expected format: list of strings
+    cl_ground_state_list = []
+    cl_ground_state_list.append(cl_ground_state)
```

Comment:
why not `cl_ground_state_list = [cl_ground_state]`?

Reply:
Changed it directly to cl_ground_state = [cl_ground_state], thanks.